galaxy/importer/finders.py

```diff
@@ -102,13 +101,18 @@ def _find_content(self):
                 yield content
 
     def _find_modules(self, content_type, content_dir):
-        for dir_path in self._module_dirs(content_dir):
-            for file_path in glob.glob(dir_path + '/*.py'):
-                file_name = os.path.basename(file_path)
-                if not os.path.isfile(file_path) or file_name == '__init__.py':
-                    continue
-                rel_path = os.path.relpath(file_path, self.path)
-                yield Result(content_type, rel_path, extra={})
+        for file_name in os.listdir(content_dir):
+            file_path = os.path.join(content_dir, file_name)
+            if os.path.isdir(file_path):
+                self.log.warning("Directory detected: '{0}'. "
+                                 "Nested modules are not supported.")
+                continue
+            if (not os.path.isfile(file_path)
+                    or not file_name.endswith('.py')
```

Comment:
Are we only importing files that end with `.py`?  Ansible modules don't have to be Python.

Reply:
Because Ansible modules are not required to be Python, we need (at some point) to add a MIME type (or some kind of indication), if possible, that we can use to tell the user what language the module is written in.

Reply:
Right, but first we should determine which file types we gonna support.